aten/src/ATen/core/ivalue.h

```diff
@@ -208,26 +202,13 @@ struct CAFFE2_API IValue final {
   }
 
   // IntList
-  IValue(c10::intrusive_ptr<ivalue::IntList> v);
   IValue(c10::ListPtr<int64_t> v);
   IValue(c10::ArrayRef<int64_t> v);
   IValue(std::vector<int64_t> v);
   bool isIntList() const { return Tag::IntList == tag; }
-  c10::intrusive_ptr<ivalue::IntList> toIntList() &&;
-  c10::intrusive_ptr<ivalue::IntList> toIntList() const &;
-
-  const c10::ListPtr<int64_t>& toIntListRef() const &;
-  c10::ListPtr<int64_t> toIntListRef() &&;
-  const c10::ListPtr<double>& toDoubleListRef() const &;
-  c10::ListPtr<double> toDoubleListRef() &&;
-  const c10::ListPtr<bool>& toBoolListRef() const &;
-  c10::ListPtr<bool> toBoolListRef() &&;
-  const c10::ListPtr<at::Tensor>& toTensorListRef() const &;
-  c10::ListPtr<at::Tensor> toTensorListRef() &&;
-  const c10::ListPtr<IValue>& toGenericListRef() const &;
-  c10::ListPtr<IValue> toGenericListRef() &&;
-  const c10::DictPtr<IValue, IValue>& toGenericDictRef() const &;
-  c10::DictPtr<IValue, IValue> toGenericDictRef() &&;
+  c10::ListPtr<int64_t> toIntList() &&;
+  c10::ListPtr<int64_t> toIntList() const &;
+
```

Comment:
so should we keep toListRef versions and make them return ArrayRef?

Reply:
We could do that. It would lock us in to the promise that we'll never unify the way we store lists (i.e. store all of them as `vector<IValue>` instead of as it is right now some as `vector<IValue>` and some as `vector<T>`), because `vector<IValue>` can't create an `ArrayRef<int64_t>`. But if I understood our last discussion correctly, this is ok?

Reply:
I added the `xxxRef` variants